DigitalLearningSolutions.Web.Tests/Controllers/LearningMenu/LearningMenuControllerTests.cs

```diff
@@ -263,6 +269,145 @@ public void Sections_should_StartOrUpdate_course_sessions()
                 .MustNotHaveHappened();
         }
 
+        [Test]
+        public void Sections_should_not_StartOrUpdate_course_sessions_if_session_not_found()
+        {
+            // Given
+            A.CallTo(() => sectionContentService.GetSectionContent(CustomisationId, CandidateId, SectionId))
+                .Returns(null);
+
+            // When
+            controller.Section(CustomisationId, SectionId);
+
+            // Then
+            A.CallTo(() => sessionService.StartOrUpdateSession(A<int>._, A<int>._, A<ISession>._)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public void Sections_should_not_StartOrUpdate_course_sessions_if_unable_to_enrol()
+        {
+            // Given
+            var defaultSectionContent = SectionContentHelper.CreateDefaultSectionContent();
+            A.CallTo(() => sectionContentService.GetSectionContent(CustomisationId, CandidateId, SectionId))
+                .Returns(defaultSectionContent);
+            A.CallTo(() => courseContentService.GetOrCreateProgressId(CandidateId, CustomisationId, CentreId))
+                .Returns(null);
+
+            // When
+            controller.Section(CustomisationId, SectionId);
+
+            // Then
+            A.CallTo(() => sessionService.StartOrUpdateSession(A<int>._, A<int>._, A<ISession>._)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public void Sections_should_UpdateProgress_course_sessions_if_valid_section()

```

Comment:
Is the `course_sessions` part of this test name necessary? I'm not quite sure what it refers to? 🙂

Reply:
It's been used in the tutorial page tests so I included it for consistency. I suppose it's more relevant to the `StartOrUpdate` tests since the `UpgradeProgress` method only uses the sessions and doesn't modify them.

Should I remove it from the tutorial tests too?

Reply:
I see! Yes, please remove them. It's only relevant to the StartOrUpdate tests 🙂